lightning/src/ln/channel.rs

```diff
@@ -5939,12 +5943,14 @@ impl<'a, Signer: Sign, K: Deref> ReadableArgs<(&'a K, u32)> for Channel<Signer>
 		// only, so we default to that if none was written.
 		let mut channel_type = Some(ChannelTypeFeatures::only_static_remote_key());
 		let mut channel_creation_height = Some(serialized_height);
+
```

Comment:
I can't leave a comment there, but on L 5717 we change the bytes read, which will break backwards compat.

`config.as_mut().unwrap().commit_upfront_shutdown_pubkey = Readable::read(reader)?;` is no longer reading a single bool, but instead an `Option<bool>` now, which is written as 2 bytes instead of 1. We should instead set it to `Some(Readable::read(reader)?);`